### ISSUE TITLE
SCC Control Library Data Lookup By Name

### ISSUE THREAD

**Issue by jor2** [IBM-Cloud/terraform-provider-ibm#5186]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

I want to be able to lookup an scc control library by name instead of id. Our terraform code creates an scc control library, and on reapply we will need to lookup the control library again by name to check for existing controls and values, but we do not have access to the `control_library_id` at that time and therefore cannot provide existing values which results in terraform removing existing controls ect. since they weren't provided.

Basically this [datasource](https://registry.terraform.io/providers/IBM-Cloud/ibm/latest/docs/data-sources/scc_profile#controls) should also accept a control library name and not have to provide an id.

### New or Affected Resource(s)

* ibm_scc_control_library

### Potential Terraform Configuration

````
data "ibm_scc_control_library" "scc_control_library" {
    instance_id = "00000000-1111-2222-3333-444444444444"
    control_library_name = "control_name"
}
````
### References

Docs: https://registry.terraform.io/providers/IBM-Cloud/ibm/latest/docs/data-sources/scc_control_library
internal issue: https://github.ibm.com/project-fortress/pm/issues/15403

* #0000

**Comment by tyao117** [comment]
https://github.com/IBM-Cloud/terraform-provider-ibm/pull/5179
